integtest/src/main/java/org/corfudb/universe/cluster/docker/CorfuServerDockerized.java

```diff
@@ -0,0 +1,324 @@
+    private final Optional<String> containerId = Optional.empty();
+    @Default
+    private final Optional<CorfuManagementServer> runtime = Optional.empty();
+
+    /**
+     * Deploys a Corfu server / docker container
+     */
+    @Override
+    public CorfuServerDockerized deploy() throws NodeException {
+        log.info("Deploying the Corfu server. {}", params.getGenericName());
+
+        if (containerId.isPresent()) {
+            throw new IllegalStateException("Corfu server already started. Parameters: " + params);
+        }
+
+        String id = deployContainer();
+
+        return CorfuServerDockerized.builder()
+                .clusterParams(clusterParams)
+                .params(params)
+                .docker(docker)
+                .containerId(Optional.of(id))
+                .runtime(Optional.of(new CorfuManagementServer(params)))
+                .build();
+    }
+
+    @Override
+    public boolean addNode(CorfuServer server) {
+        return runtime.map(rt -> rt.add(server)).orElse(false);
+    }
+
+    @Override
+    public boolean removeNode(CorfuServer server) {
+        return runtime.map(rt -> rt.remove(server)).orElse(false);
+    }
+
+
+    @Override
+    public Optional<Layout> getLayout() {
+        return runtime.map(CorfuManagementServer::getLayout);
+    }
+
+    @Override
+    public void connectCorfuRuntime() {
+        runtime.ifPresent(CorfuManagementServer::connect);
+    }
+
+    /**
+     * This method attempts to gracefully stops the Corfu server. After timeout, it will kill the Corfu server.
+     *
+     * @param timeout a duration after which the stop will kill the server
+     * @throws NodeException this exception will be thrown if the server can not be stopped.
+     */
+    @Override
+    public void stop(Duration timeout) throws NodeException {
+        log.info("Stopping the Corfu server. {}", params.getGenericName());
+
+        try {
+            docker.stopContainer(params.getGenericName(), (int) timeout.getSeconds());
+        } catch (DockerException | InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new NodeException("Can't stop Corfu server", e);
+        }
+    }
+
+    /**
+     * Immediately kill the Corfu server.
+     *
+     * @throws NodeException this exception will be thrown if the server can not be killed.
+     */
+    @Override
+    public void kill() throws NodeException {
+        log.info("Killing the Corfu server. {}", params.getGenericName());
+
+        try {
+            docker.killContainer(params.getGenericName());
+        } catch (DockerException | InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            throw new NodeException("Can't kill Corfu server", ex);
+        }
+    }
+
+    /**
+     * Deploy and start docker container, expose ports, connect to a network
+     *
+     * @return docker container id
+     */
+    private String deployContainer() throws NodeException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.universe.node.NodeException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)